frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx

```diff
@@ -0,0 +1,488 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
+import { Form, FormControl, FormGroup, HelpBlock } from 'patternfly-react';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { ButtonBar, Dropdown, Firehose, history } from '@console/internal/components/utils';
+import { k8sCreate } from '@console/internal/module/k8s';
+import { validateDNS1123SubdomainValue, ValidationErrorType } from '@console/shared';
+import { NetworkAttachmentDefinitionModel, SriovNetworkNodePolicyModel } from '../..';
+import { NetworkAttachmentDefinitionConfig } from '../../types';
+import { ELEMENT_TYPES, NetworkTypeParams, networkTypeParams, networkTypes } from '../../constants';
+
+const buildConfig = (name, networkType, typeParamsData): NetworkAttachmentDefinitionConfig => {
+  const config: NetworkAttachmentDefinitionConfig = {
+    name,
+    type: networkType,
+    cniVersion: '0.3.1',
+  };
+
+  _.forOwn(typeParamsData, (val, key) => {
+    config[key] = _.get(val, 'value', null);
+  });
+
+  return config;
+};
+
+const createNetAttachDef = (
+  e: React.FormEvent<EventTarget>,
+  description,
+  name,
+  networkType,
+  typeParamsData,
+  namespace,
+  setError,
+  setLoading,
+) => {
+  e.preventDefault();
+
+  setLoading(true);
+
+  const config = JSON.stringify(buildConfig(name, networkType, typeParamsData));
+  const newNetAttachDef = {
+    apiVersion: 'k8s.cni.cncf.io/v1',
```

Comment:
it would be better to get this from `NetworkAttachmentDefinitionModel` such as NetworkAttachmentDefinitionModel.apiGroup/NetworkAttachmentDefinitionModel.apiVersion

Reply:
Done!